### DISCUSSION TITLE
Increase page navigation flexabiltiy

### DISCUSSION THREAD

**Discussion by billybonks** [teamcapybara/capybara#2753] [Ideas]
There is limited capability to navigate the page using HTML semantics

For example find **Section A** using **Heading text** within it assert **button**

Using existing apis all you can do is, Use Css selectors to find the section then use another css selector to assert the heading, but there is a semantic way to interact with the button.

for example 

```
find('div#delivery-address').fill_in('Street', with: '12 Main Street')
```

Would become

```
find_section('Delivery Address').fill_in('Street', with: '12 Main Street')
```

I think the semantic ways of interacting with intractable components are awesome and the dream would be to be able to navigate the page using those semantics instead of css selectors.

css will probably change first before the semantics of a page, wanted to star this discussion first to see sentiment before going further with building/designing  anything

**Comment by billybonks** [comment]
somewhat related thread 

https://github.com/orgs/teamcapybara/discussions/2720

**Comment by twalpole** [comment]
And how would you define  a 'section'? IF it varies with every web framework then it's not going to be implemented in Capybara. Seems more like it would be a separate library built on top of Capybara to work with a specific framework
